Octokit.Tests.Integration/Clients/RepositoryBranchesClientTests.cs

```diff
@@ -543,4 +544,999 @@ public TheDeleteBranchProtectionMethod()
+        RepositoryContext _userRepoContext;
+        OrganizationRepositoryWithTeamContext _orgRepoContext;
+
+        public TheUpdateRequiredStatusChecksContextsMethod()
+        {
+            var github = Helper.GetAuthenticatedClient();
+            _client = github.Repository.Branch;
+
+            _userRepoContext = github.CreateRepositoryWithProtectedBranch().Result;
+            _orgRepoContext = github.CreateOrganizationRepositoryWithProtectedBranch().Result;
+        }
+
+        [IntegrationTest]
+        public async Task UpdateRequiredStatusChecksContexts()
+        {
+            var repoOwner = _userRepoContext.RepositoryOwner;
+            var repoName = _userRepoContext.RepositoryName;
+            var update = new List<string>() { "newMember1", "newMember2" };
+            var requiredStatusChecksContexts = await _client.UpdateRequiredStatusChecksContexts(repoOwner, repoName, "master", update);
+
+            Assert.NotNull(requiredStatusChecksContexts);
+            Assert.NotNull(requiredStatusChecksContexts.Count);
+            Assert.Equal(2, requiredStatusChecksContexts.Count);
+        }
+
+        [IntegrationTest]
+        public async Task UpdatesRequiredStatusChecksContextsWithRepositoryId()
+        {
+            var repoId = _userRepoContext.RepositoryId;
+            var update = new List<string>() { "newMember1", "newMember2" };
+            var requiredStatusChecksContexts = await _client.UpdateRequiredStatusChecksContexts(repoId, "master", update);
+
+            Assert.NotNull(requiredStatusChecksContexts);
+            Assert.NotNull(requiredStatusChecksContexts.Count);
+            Assert.Equal(2, requiredStatusChecksContexts.Count);
+        }
+
+        [IntegrationTest]
+        public async Task UpdatesRequiredStatusChecksContextsForOrgRepo()
+        {
+            var repoOwner = _orgRepoContext.RepositoryContext.RepositoryOwner;
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+            var update = new List<string>() { "newMember1", "newMember2" };
+            var requiredStatusChecksContexts = await _client.UpdateRequiredStatusChecksContexts(repoOwner, repoName, "master", update);
+
+            Assert.NotNull(requiredStatusChecksContexts);
+            Assert.NotNull(requiredStatusChecksContexts.Count);
+            Assert.Equal(2, requiredStatusChecksContexts.Count);
+        }
+
+        [IntegrationTest]
+        public async Task UpdatesRequiredStatusChecksContextsForOrgRepoWithRepositoryId()
+        {
+            var repoId = _orgRepoContext.RepositoryContext.RepositoryId;
+            var update = new List<string>() { "newMember1", "newMember2" };
+            var requiredStatusChecksContexts = await _client.UpdateRequiredStatusChecksContexts(repoId, "master", update);
+
+            Assert.NotNull(requiredStatusChecksContexts);
+            Assert.NotNull(requiredStatusChecksContexts.Count);
+            Assert.Equal(2, requiredStatusChecksContexts.Count);
```

Comment:
How about making this method UPDATE the status checks to a different number of contexts than they started with.  eg there were 2 to start with, so make this test Update it to only 1.  Otherwise perhaps this method didnt do anyting and the count of 2 is juts the original 2 still there?  :grinning: